src/components/authorization/authorization.service.ts

```diff
@@ -104,6 +109,37 @@ export class AuthorizationService {
     return parseSecuredProperties(props, permissions, resource.SecuredProps);
   }
 
+  async verifyCanEditChanges<
+    TObject extends Resource,
+    Key extends keyof TObject & string
+  >(
+    baseNode: TObject,
+    props: readonly Key[],
+    changes: { [Key in keyof TObject]?: UnwrapSecured<TObject[Key]> }
+  ) {
+    for (const prop of props) {
+      if (
+        changes[prop] === undefined ||
+        unwrapSecured(baseNode[prop]) === changes[prop]
+      ) {
+        continue;
+      }
+      await this.verifyCanEdit(baseNode, prop);
+    }
+  }
+
+  async verifyCanEdit<
+    DbNode extends Record<string, any>,
+    Key extends keyof DbNode
+  >(baseNode: DbNode, prop: Key) {
+    if (isSecured(baseNode[prop]) && !baseNode[prop].canEdit) {
+      throw new UnauthorizedException(
+        'You do not have permission to update this property',
```

Comment:
The message could reference the object & property to be more helpful.
```suggestion
        'You do not have permission to update ${resource.name}.${prop}',
```